lib/constants/encoding_charset.js

```diff
@@ -3,45 +3,44 @@
 
 module.exports = {
   big5: 1,
-  latin2: 2,
   dec8: 3,
   cp850: 4,
-  latin1: 5,
   hp8: 6,
   koi8r: 7,
+  latin1: 8,
+  latin2: 9,
   swe7: 10,
   ascii: 11,
   ujis: 12,
   sjis: 13,
-  cp1251: 14,
   hebrew: 16,
   tis620: 18,
   euckr: 19,
-  latin7: 20,
   koi8u: 22,
   gb2312: 24,
   greek: 25,
   cp1250: 26,
   gbk: 28,
-  cp1257: 29,
   latin5: 30,
   armscii8: 32,
-  cesu8: 33,
+  utf8: 33,
```

Comment:
mysql utf8 is in fact 'cesu8' encoding ( while what they call 'utf8mb4' is normal utf8 ), that is why cesu8 was there originally. It would be good to have more tests to validate this, see some discussion here - https://github.com/sidorares/node-mysql2/pull/374#issuecomment-241728060